File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
answer = 0
alphabets = ['A','E','I','O','U']

def dfs(string, word):
    global answer, alphabets
    answer += 1
    
    if (word == string):
        return True
    
    if len(string) == 5:
        return False
    
    for alpha in alphabets:
        if dfs(string + alpha, word) == True:
            return True

def solution(word):

    for alpha in alphabets:
        if dfs(alpha, word) == True:
            return answer
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let words = []
    const alphabet = ['A','E','I','O','U']
    
    function dfs(cur) {
        if (cur.length === 5) return false
        
        for (let alpha of alphabet) {
            let newWord = cur + alpha
            words.push(newWord)
            dfs(newWord)
        }
    }
    
    dfs('')
    
    return words.indexOf(word) + 1
    
}